src/policy/sft_map.rs

```diff
@@ -186,21 +194,15 @@ mod space_map {
             start: Address,
             bytes: usize,
         ) {
-            let table_size = Self::addr_to_index(Address::MAX) + 1;
             let index = Self::addr_to_index(start);
             if cfg!(debug_assertions) {
                 // Make sure we only update from empty to a valid space, or overwrite the space
                 let old = self.sft[index].load();
                 assert!((*old).name() == EMPTY_SFT_NAME || (*old).name() == (*space).name());
                 // Make sure the range is in the space
                 let space_start = Self::index_to_space_start(index);
-                // FIXME: Curerntly skip the check for the last space. The following works fine for MMTk internal spaces,
-                // but the VM space is an exception. Any address after the last space is considered as the last space,
-                // based on our indexing function. In that case, we cannot assume the end of the region is within the last space (with MAX_SPACE_EXTENT).
-                if index != table_size - 1 {
-                    assert!(start >= space_start);
-                    assert!(start + bytes <= space_start + vm_layout().max_space_extent());
-                }
+                assert!(start >= space_start);
+                assert!(start + bytes <= space_start + vm_layout().max_space_extent(), "The range of {} + {} bytes does not fall into the space range {} and {}, and it is probably outside the address range we use.", start, bytes, space_start, space_start + vm_layout().max_space_extent());
```

Comment:
I am surprised that `cargo fmt` simply ignores this line.  It's likely that the string literal is too long.  After breaking the string literal so that it can fit into 100-character line width, `cargo fmt` formats it properly.

And it is more concise and more accurate to use the half-open interval notation `[start_inclusive, end_exclusive)`

```suggestion
                assert!(
                    start + bytes <= space_start + vm_layout().max_space_extent(),
                    "The range [{}, {}) ({} bytes) does not fall into the space range [{}, {}), \
                    and it is probably outside the address range we use.",
                    start,
                    start + bytes,
                    bytes,
                    space_start,
                    space_start + vm_layout().max_space_extent()
                );
 ```

Reply:
Sounds good.